src/components/include/utils/threads/thread.h

```diff
@@ -169,15 +192,20 @@ class Thread {
    * be used as if it were newly constructed (i.e., Start may be called again).
    *
    * Stop may be called multiple times and is simply ignored if the thread is
-   * already stopped.
+   * already stopped. Thread safe.
    */
   void stop();
 
   /**
    * @brief Starts join procedure.
-   * @param[in] force_stop If true stop() is called first otherwise is skipped.
+   * @param[in] force_stop for controlling call of stop() method first.
+   * If it is true stop() is called and stopped_ flag is set to true in order
+   * to command/force the thread to stop.
+   * If it is false stop() method is not called and stopped_ flag is not set.
+   * It is up to the thread delegate when/how will end.
+   * In both cases join() waits thread delegate to finish.
    */
-  void join(const bool force_stop = true);
+  void join(const bool force_stop);
```

Comment:
@anosach-luxoft, there was a discussion with @pestOO (including some emails I sent with you in cc - check "My findings after brief review of thread_posix.cc regarding APPLINK-25909").
In general the question was is this sequence is legal by design:
thread->start();
thread->join();
ASSERT(job_is_done_successfully /*despite_immediate_stop?*/);
Right now you can find it only in unit tests, but the decision was to introduce this flag in join() and make above sequence legal.